### PR TITLE
PS4 report under new framework

### PR DESCRIPTION
**Pull request by zduey**
Implements the logic necessary to build PS4 session reports. Primary changes/implementations include:
- Special section in umbrella make_report task since PS4 shares so little with other stim reports
- Created a new PSStimSessionSummary object with properties for extracting data needed for plotting
- Updated data structure passed to PS4 plotting functions

### CODE REVIEW

File: ramutils/pipelines/report.py [hunk 1]
```diff
@@ -63,13 +63,16 @@ def make_report(subject, experiment, paths, joint_report=False,
...
 
-    # all_events are used for producing math summaries. Task events are only
-    # used by the stim reports. Non-stim reports create a different set of
-    # events. Stim params are used in building the stim session summaries
-    all_events, task_events, stim_params = build_test_data(subject, experiment,
-                                                           paths, joint_report,
-                                                           sessions=sessions,
-                                                           **kwargs)
+    if 'PS' not in experiment:
+        # all_events are used for producing math summaries. Task events are only
```
Comment (mivade) [inline ramutils/pipelines/report.py]: Can you add a comment about why specifically PS is excluded here?

File: ramutils/reports/generate.py [hunk 2]
```diff
@@ -59,15 +59,16 @@ def __init__(self, session_summaries, math_summaries,
                               self.session_summaries]
         self.catfr_summaries = list(compress(self.session_summaries, catfr_summary_mask))
 
-        if len(session_summaries) != len(math_summaries):
-            raise ValueError("Summaries contain different numbers of sessions")
-
+        # FIXME: PS4 has no math summaries, so this check won't work
```
Comment (mivade) [inline ramutils/reports/generate.py]: I'd keep this check in for now, conditioned on it not being a PS4 experiment.

File: ramutils/reports/static/plots.js [hunk 3]
```diff
@@ -305,73 +305,93 @@ var ramutils = (function (mod, Plotly) {
...
             {
-              x: amplitude,
-              y: retrieval[labels[i]],
+              x: data_dict[labels[i]]['amplitude']['RETRIEVAL'],
+              y: data_dict[labels[i]]['delta_classifier']['RETRIEVAL'],
               mode: 'markers',
               name: 'Retrieval',
+              xaxis: xaxis,
+            }
+            // Add back once we actually produce sham data results
```
Comment (mivade) [inline ramutils/reports/static/plots.js]: Please prepend with FIXME or TODO (PyCharm, VS Code, and others have support for listing things marked this way which we should probably take advantage of).

File: ramutils/reports/static/plots.js [hunk 4]
```diff
@@ -305,73 +305,93 @@ var ramutils = (function (mod, Plotly) {
...
-              x: amplitude,
-              y: sham[labels[i]],
+              x: data_dict[labels[i]]['post_stim_amplitude']['RETRIEVAL'],
+              y: data_dict[labels[i]]['post_stim_biomarker']['RETRIEVAL'],
               mode: 'markers',
-              name: 'Sham',
+              name: 'Retrieval',
               xaxis: xaxis
             }
+            // Add back once we actually produce sham data results
```
Comment (mivade) [inline ramutils/reports/static/plots.js]: Same comment as above.

**Comment by zduey** [comment]
Code changes based on the provided comments will be included in the next PR along with some other final changes before the master branch is release-ready

**Pull request merged**